lib/ts/recipe/thirdpartyemailpassword/emaildelivery/services/backwardCompatibility/index.ts

```diff
@@ -0,0 +1,93 @@
+import { User as EmailVerificationUser } from "../../../../emailverification/types";
+import { NormalisedAppinfo } from "../../../../../types";
+import EmailVerificationBackwardCompatibilityService from "../../../../emailverification/emaildelivery/services/backwardCompatibility";
+import EmailPasswordBackwardCompatibilityService from "../../../../emailpassword/emaildelivery/services/backwardCompatibility";
+import { EmailDeliveryInterface } from "../../../../../ingredients/emaildelivery/types";
+
+export default class BackwardCompatibilityService
+    implements EmailDeliveryInterface<TypeThirdPartyEmailPasswordEmailDeliveryInput> {
+    private emailPasswordBackwardCompatibilityService: EmailPasswordBackwardCompatibilityService;
+    private emailVerificationBackwardCompatibilityService: EmailVerificationBackwardCompatibilityService;
+
+    constructor(
+        recipeInterfaceImpl: RecipeInterface,
+        emailPasswordRecipeInterfaceImpl: EmailPasswordRecipeInterface,
+        appInfo: NormalisedAppinfo,
+        isInServerlessEnv: boolean,
+        resetPasswordUsingTokenFeature?: {
+            createAndSendCustomEmail?: (
+                user: User,
+                passwordResetURLWithToken: string,
+                userContext: any
+            ) => Promise<void>;
+        },
+        emailVerificationFeature?: {
+            createAndSendCustomEmail?: (
+                user: User,
+                emailVerificationURLWithToken: string,
+                userContext: any
+            ) => Promise<void>;
+        }
+    ) {
+        {
+            const inputCreateAndSendCustomEmail = emailVerificationFeature?.createAndSendCustomEmail;
+            let emailVerificationFeatureNormalisedConfig =
+                inputCreateAndSendCustomEmail !== undefined
+                    ? {
+                          createAndSendCustomEmail: async (
+                              user: EmailVerificationUser,
+                              link: string,
+                              userContext: any
+                          ) => {
+                              let userInfo = await recipeInterfaceImpl.getUserById({
+                                  userId: user.id,
+                                  userContext,
+                              });
+                              if (userInfo === undefined) {
+                                  throw new Error("Unknown User ID provided");
+                              }
+                              return await inputCreateAndSendCustomEmail(userInfo, link, userContext);
+                          },
+                      }
+                    : {};
+            this.emailVerificationBackwardCompatibilityService = new EmailVerificationBackwardCompatibilityService(
+                appInfo,
+                isInServerlessEnv,
+                emailVerificationFeatureNormalisedConfig.createAndSendCustomEmail
+            );
```

Comment:
You do not need to make this. Just emailPasswordBackwardCompatibilityService should be enough.